highlighting/lib/languages/dart.dart

```diff
@@ -3,215 +3,289 @@
-          relevance: 0),
-      Mode(scope: "comment", begin: "\\/{3,} ?", end: "\$", contains: [
-        Mode(subLanguage: ["markdown"], begin: ".", end: "\$", relevance: 0),
-        Mode(
-            scope: "doctag",
-            begin: "[ ]*(?=(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):)",
-            end: "(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):",
-            excludeBegin: true,
-            relevance: 0),
-        Mode(
-            begin:
-                "[ ]+((?:I|a|is|so|us|to|at|if|in|it|on|[A-Za-z]+['](d|ve|re|ll|t|s|n)|[A-Za-z]+[-][a-z]+|[A-Za-z][a-z]{2,})[.]?[:]?([.][ ]|[ ])){3}")
-      ]),
-      C_LINE_COMMENT_MODE,
-      C_BLOCK_COMMENT_MODE,
-      Mode(
-          className: "class",
-          beginKeywords: "class interface",
-          end: "\\{",
-          excludeEnd: true,
+        ),
+        Mode(
+          begin: "\"\"\"",
+          end: "\"\"\"",
+          contains: [
+            BACKSLASH_ESCAPE,
+            Mode(ref: '~contains~0~variants~4~contains~1'),
+            Mode(ref: '~contains~0~variants~4~contains~2'),
+          ],
+        ),
+        Mode(
+          begin: "'",
+          end: "'",
+          illegal: "\\n",
+          contains: [
+            BACKSLASH_ESCAPE,
+            Mode(ref: '~contains~0~variants~4~contains~1'),
+            Mode(ref: '~contains~0~variants~4~contains~2'),
+          ],
+        ),
+        Mode(
+          begin: "\"",
+          end: "\"",
+          illegal: "\\n",
           contains: [
-            Mode(beginKeywords: "extends implements"),
-            UNDERSCORE_TITLE_MODE
-          ]),
-      C_NUMBER_MODE,
-      Mode(className: "meta", begin: "@[A-Za-z]+"),
-      Mode(begin: "=>")
-    ]);
+            BACKSLASH_ESCAPE,
+            Mode(ref: '~contains~0~variants~4~contains~1'),
+            Mode(ref: '~contains~0~variants~4~contains~2'),
+          ],
+        ),
+      ],
+    ),
+  },
+  name: "Dart",
+  keywords: {
+    "keyword": [
+      "abstract",
+      "as",
+      "assert",
+      "async",
+      "await",
+      "break",
+      "case",
+      "catch",
+      "class",
+      "const",
+      "continue",
+      "covariant",
+      "default",
+      "deferred",
+      "do",
+      "dynamic",
+      "else",
+      "enum",
+      "export",
+      "extends",
+      "extension",
+      "external",
+      "factory",
+      "false",
+      "final",
+      "finally",
+      "for",
+      "Function",
+      "get",
+      "hide",
+      "if",
+      "implements",
+      "import",
+      "in",
+      "inferface",
+      "is",
+      "late",
+      "library",
+      "mixin",
+      "new",
+      "null",
+      "on",
+      "operator",
+      "part",
+      "required",
+      "rethrow",
+      "return",
+      "set",
+      "show",
+      "static",
+      "super",
+      "switch",
+      "sync",
+      "this",
+      "throw",
+      "true",
+      "try",
+      "typedef",
+      "var",
+      "void",
+      "while",
+      "with",
+      "yield"
```

Comment:
Not complete yet.

Reply:
Keywords are dumped with `JSON.stringify()` which does not add trailing commas. We need a replacement to it that does. There must be 3rd party packages for this. This may also reduce special cases elsewhere, like `arr.length === 0` in `variants`.